Content.Shared/Revolutionary/HeadRevolutionaryComponent.cs

```diff
@@ -0,0 +1,13 @@
+using Robust.Shared.GameStates;
+using Content.Shared.StatusIcon;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.Revolutionary;
+
+[RegisterComponent, NetworkedComponent]
+
+public sealed class HeadRevolutionaryComponent : Component
+{
+    [DataField("headRevStatusIcon", customTypeSerializer: typeof(PrototypeIdSerializer<StatusIconPrototype>))]
+    public string HeadRevStatusIcon = "HeadRevolutionaryFaction";
+}
```

Comment:
1. Access
2. Formatting
3. Documenting.
4. This must use player-only states.

Reply:
Can you explain the issues with the access, formatting, and what a player-only state is?

Reply:
Access is an attribute you can add to a class to prevent other systems from using it except through an explicitly declared public API.

For example: `[Access(typeof(NukeSystem))]`

Reply:
I think flashcomp and some others use it, but it prevents leaking the component's existence to cheaters.

Reply:
I tried it yesterday Vorden (also thanks for your help on these) but the main issue I am having is that the system that uses this component doesn't pop up and I guess it's because it's clientside? I'm not really sure but I could use some help.

Reply:
Networking a component while also restricting which players can receive it requires a bit more work. I have a draft branch that adds some systems to manage (non-npc) faction membership networking. Ill probably open that as a PR sometime soon, then using that is probably the easiest solution.